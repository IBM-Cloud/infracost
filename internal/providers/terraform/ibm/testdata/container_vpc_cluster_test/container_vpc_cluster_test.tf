terraform {
  required_providers {
    ibm = {
      source = "IBM-Cloud/ibm"
    }
  }
}
provider "ibm" {
  region           = "us-south"
  ibmcloud_timeout = "1"
  max_retries      = "1"
}

resource "ibm_is_vpc" "vpc1" {
  name = "myvpc"
}

resource "ibm_is_subnet" "subnet1" {
  name                     = "mysubnet1"
  vpc                      = ibm_is_vpc.vpc1.id
  zone                     = "us-south-1"
  total_ipv4_address_count = 256
}

resource "ibm_is_subnet" "subnet2" {
  name                     = "mysubnet2"
  vpc                      = ibm_is_vpc.vpc1.id
  zone                     = "us-south-2"
  total_ipv4_address_count = 256
}

resource "ibm_container_vpc_cluster" "cluster" {
  name         = "mycluster"
  vpc_id       = ibm_is_vpc.vpc1.id
  flavor       = "bx2.4x16"
  worker_count = 3
  kube_version = "1.17.5"
  zones {
    subnet_id = ibm_is_subnet.subnet1.id
    name      = "us-south-1"
  }
  zones {
    subnet_id = ibm_is_subnet.subnet1.id
    name      = "us-south-2"
  }
}

resource "ibm_container_vpc_cluster" "cluster_without_usage" {
  name         = "mycluster-without-usage"
  vpc_id       = ibm_is_vpc.vpc1.id
  flavor       = "bx2.4x16"
  worker_count = 3
  kube_version = "1.17.5"
  zones {
    subnet_id = ibm_is_subnet.subnet1.id
    name      = "us-south-1"
  }
  zones {
    subnet_id = ibm_is_subnet.subnet1.id
    name      = "us-south-2"
  }
}

resource "ibm_container_vpc_cluster" "roks_cluster_with_usage" {
  name         = "myrokscluster-with-usage"
  vpc_id       = ibm_is_vpc.vpc1.id
  flavor       = "bx2.4x16"
  worker_count = 3
  kube_version = "4.13_openshift"
  zones {
    subnet_id = ibm_is_subnet.subnet1.id
    name      = "us-south-1"
  }
  zones {
    subnet_id = ibm_is_subnet.subnet1.id
    name      = "us-south-2"
  }
}

resource "ibm_container_vpc_cluster" "roks_with_entitlement" {
  name         = "roks-with-entitlement"
  vpc_id       = ibm_is_vpc.vpc1.id
  flavor       = "bx2.4x16"
  worker_count = 3
  kube_version = "4.13_openshift"
  entitlement  = "cloud_pak"
  zones {
    subnet_id = ibm_is_subnet.subnet1.id
    name      = "us-south-1"
  }
  zones {
    subnet_id = ibm_is_subnet.subnet1.id
    name      = "us-south-2"
  }
}

resource "ibm_container_vpc_cluster" "roks_no_entitlement" {
  name         = "roks-no-entitlement"
  vpc_id       = ibm_is_vpc.vpc1.id
  flavor       = "bx2.4x16"
  worker_count = 3
  kube_version = "4.13_openshift"
  zones {
    subnet_id = ibm_is_subnet.subnet1.id
    name      = "us-south-1"
  }
  zones {
    subnet_id = ibm_is_subnet.subnet1.id
    name      = "us-south-2"
  }
}

/*
  Copies this OpenShift container platform on VPC configuration: 
  https://cloud.ibm.com/catalog/7a4d68b4-cf8b-40cd-a3d1-f49aff526eb3/architecture/deploy-arch-ibm-ocp-vpc-1728a4fd-f561-4cf9-82ef-2b1eeb5da1a8-global

  Uses 2 clusters and 2 worker pools each running for 730 hours.
*/
resource "ibm_container_vpc_cluster" "roks_cluster_vpc" {
  name         = "roks-cluster-vpc"
  vpc_id       = ibm_is_vpc.vpc1.id
  flavor       = "bx2.8x32"
  worker_count = 2
  kube_version = "4.17_openshift" // Version as shown in console
  zones {
    subnet_id = ibm_is_subnet.subnet1.id
    name      = "us-south-1"
  }
}

resource "ibm_container_vpc_worker_pool" "roks_worker_pool" {
  cluster          = ibm_container_vpc_cluster.roks_cluster_vpc.id
  worker_pool_name = "mywp"
  flavor           = "bx2.8x32"
  vpc_id           = ibm_is_vpc.vpc1.id
  worker_count     = 2
  entitlement      = "cloud-pak"
  zones {
    name      = "us-south-2"
    subnet_id = ibm_is_subnet.subnet2.id
  }
}