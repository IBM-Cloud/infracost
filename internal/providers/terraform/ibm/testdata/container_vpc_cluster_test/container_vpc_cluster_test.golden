
 Name                                                                                              Monthly Qty  Unit      Monthly Cost 
                                                                                                                                       
 ibm_is_vpc.vpc1                                                                                                                       
 ├─ VPC instance                                                                                             1  Instance         $0.00 
 ├─ VPC egress free allowance (first 5GB)                                                                    5  GB               $0.00 
 └─ VPC egress us-south (first 9995 GB)                                                                  9,995  GB             $908.70 
 └─ VPC egress us-south (next 40000 GB)                                                                 40,000  GB           $3,469.40 
 └─ VPC egress us-south (next 100000 GB)                                                                   100  GB               $7.32 
 └─ VPC egress us-south (over 149995 GB)                                                                     0  GB               $0.00 
                                                                                                                                       
 mycluster                                                                                                                             
 ├─ VPC Container Work Zone flavor: (bx2.4x16) region: (us-south) name: (us-south-1) x(3) workers        2,190  hours          $608.59 
 └─ VPC Container Work Zone flavor: (bx2.4x16) region: (us-south) name: (us-south-2) x(3) workers        2,190  hours          $608.59 
                                                                                                                                       
 mycluster-without-usage                                                                                                               
 ├─ VPC Container Work Zone flavor: (bx2.4x16) region: (us-south) name: (us-south-1) x(3) workers            3  hours            $0.83 
 └─ VPC Container Work Zone flavor: (bx2.4x16) region: (us-south) name: (us-south-2) x(3) workers            3  hours            $0.83 
                                                                                                                                       
 myrokscluster-with-usage                                                                                                               
 ├─ VPC Container Work Zone flavor: (bx2.4x16) region: (us-south) name: (us-south-1) x(3) workers        2,190  hours          $983.08 
 └─ VPC Container Work Zone flavor: (bx2.4x16) region: (us-south) name: (us-south-2) x(3) workers        2,190  hours          $983.08 
                                                                                                                                       
 OVERALL TOTAL                                                                                                               $7,570.40 
──────────────────────────────────
6 cloud resources were detected:
∙ 4 were estimated, 1 of which usage-based costs, see https://infracost.io/usage-file
∙ 2 were free:
  ∙ 2 x ibm_is_subnet